src/ORM/Table.php

```diff
@@ -2304,4 +2305,28 @@ public function __debugInfo()
             'connectionName' => $conn ? $conn->configName() : null
         ];
     }
+
+    /**
+     * Validates that a belongsToMany or hasMany assoc has at least one entry.
+     *
+     * This can be used to validate any other array as well.
+     *
+     * @param mixed $value
+     * @return boolean
+     */
+    public function validateCount($value, $expectedCount = 0, $operator = '>')
```

Comment:
This feels like an application rule to me and not a request validator. I am guessing that you want to limit the number of records an association has which is more similar to the UniqueRule.